modin/numpy/arr.py

```diff
@@ -0,0 +1,1459 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+"""Module houses ``array`` class, that is distributed version of ``numpy.array``."""
+
+from math import prod
+import numpy
+from pandas.core.dtypes.common import is_list_like, is_numeric_dtype, is_bool_dtype
+from pandas.api.types import is_scalar
+from inspect import signature
+
+import modin.pandas as pd
+from modin.error_message import ErrorMessage
+from modin.core.dataframe.algebra import (
+    Map,
+    Reduce,
+    Binary,
+)
```

Comment:
Ooh, we really mix up the Modin Layers here. I want us to get rid of this ASAP in follow-up PRs and implement the proper structure.

Reply:
That's definitely true - we are considering on adding NumPy partitioning, etc. so when those PRs go up, we'll be sure to refactor this so it doesn't broach layers!